### PR TITLE
add rolling upgrade test to kola

### PR DESCRIPTION
**Pull request by pbx0**
/cc @marineam @yichengq 

The test works by starting a 2.0.12 etcd cluster. Then a rolling upgrade is started by bringing down and replacing the binary of one node at a time. During this , random writes are being made and checked to try and ensure the cluster remains available the entire time. The specific versions are configurable, but right now it tests 2.0.12 -> 2.1.0-alpha.1 using precompiled binaries. As far as this test can tell, the reliable works and the final cluster version is updated.

@marineam I added a bit of a hack to get some flags into the rolling upgrade test itself.

### CODE REVIEW

**Comment by marineam** [comment]
I'd rather not check in binaries into git

**Comment by pbx0** [comment]
Removed the checked in binaries and changed the default location to look for them to just be `etcd` and `etcd2` in the working dir. 

In making this work more generically across any etcd binaries that support rolling updates, I added a skipVersionCheck flag so you don't have to specify the  versions you expect out of etcd. Just the rolling update, health checks, and availability are checked.

**Comment by marineam** [comment]
LGTM

File: kola/tests/etcd/rolling.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		for _, m := range cluster.Machines() {
+			if err := m.StartJournal(); err != nil {
+				return fmt.Errorf("failed to start journal: %v", err)
+			}
+		}
+	}
+
+	// drop in starting etcd binary
+	plog.Debug("adding files to cluster")
+	if err := cluster.DropFile(etcdBin); err != nil {
```
Comment (marineam) [inline kola/tests/etcd/rolling.go]: perhaps add a TODO to gracefully skip tests if the binaries aren't available once there is a meaningful way to actually do that.

Comment (pbx0) [inline kola/tests/etcd/rolling.go]: done

**Pull request merged**